rest/model/pod.go

```diff
@@ -0,0 +1,77 @@
+package model
+
+import (
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/pod"
+	"github.com/evergreen-ci/utility"
+)
+
+// APISecretOpts is the model for secrets in a container.
+type APISecretOpts struct {
+	Name   *string
+	Value  *string
+	Owned  *bool
+	Exists *bool
+}
+
+// APIPodEnvVar is the model for environment variables in a container
+type APIPodEnvVar struct {
+	Name       *string
+	Value      *string
+	SecretOpts *APISecretOpts
+}
+
+// APITimeInfo is the model for the timing information of a pod.
+type APITimeInfo struct {
+	Initialized time.Time
+	Started     time.Time
+	Provisioned time.Time
+}
+
+// APIPod is the model to be returned by the API whenever pod.Pod is fetched.
+type APIPod struct {
+	Name     *string
```

Comment:
All the fields need JSON struct tags so that the API knows how to parse the JSON from the request we send to the REST route into this struct. See the comments in [json.Marshal](https://pkg.go.dev/encoding/json#Marshal) about struct tags.